matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/room/aggregation/livelocation/LiveLocationAggregationProcessor.kt

```diff
@@ -56,8 +67,39 @@ internal class LiveLocationAggregationProcessor @Inject constructor() {
 
         Timber.d("updating summary of id=$targetEventId with isLive=${content.isLive}")
 
-        aggregatedSummary.endOfLiveTimestampMillis = content.getBestTimestampMillis()?.let { it + (content.timeout ?: 0) }
-        aggregatedSummary.isActive = content.isLive
+        val endOfLiveTimestampMillis = content.getBestTimestampMillis()?.let { it + (content.timeout ?: 0) }
+        aggregatedSummary.endOfLiveTimestampMillis = endOfLiveTimestampMillis
+        aggregatedSummary.isActive = isLive
+
+        if (isLive) {
+            scheduleDeactivationAfterTimeout(targetEventId, roomId, endOfLiveTimestampMillis)
+        } else {
+            cancelDeactivationAfterTimeout(targetEventId, roomId)
+        }
+    }
+
+    private fun scheduleDeactivationAfterTimeout(eventId: String, roomId: String, endOfLiveTimestampMillis: Long?) {
+        endOfLiveTimestampMillis?.let { endOfLiveMillis ->
+            val workParams = DeactivateLiveLocationShareWorker.Params(sessionId = sessionId, eventId = eventId, roomId = roomId)
+            val workData = WorkerParamsFactory.toData(workParams)
+            val workName = DeactivateLiveLocationShareWorker.getWorkName(eventId = eventId, roomId = roomId)
+            val workDelayMillis = (endOfLiveMillis - clock.epochMillis()).coerceAtLeast(0)
+            val workRequest = workManagerProvider.matrixOneTimeWorkRequestBuilder<DeactivateLiveLocationShareWorker>()
+                    .setInitialDelay(workDelayMillis, TimeUnit.MILLISECONDS)
+                    .setInputData(workData)
+                    .build()
+
+            workManagerProvider.workManager.enqueueUniqueWork(
+                    workName,
+                    ExistingWorkPolicy.KEEP,
```

Comment:
I think you should use `ExistingWorkPolicy.REPLACE` here.

Reply:
I have done the change in [8864a3c](https://github.com/vector-im/element-android/pull/6128/commits/8864a3cf2c460964f4e63f74c4c6e1a6ea4f1fb2). But I am not sure, it will make a big difference.